### ISSUE TITLE
Visualizing class activation maps with tflite weights

### ISSUE THREAD

**Issue by ra9hur** [tensorflow/tensorflow#58468]
I am able to visualize Grad-CAM using keras weights(H5). I am not able to find to visualize last CNN layer using tflite weights.

- https://pyimagesearch.com/2020/03/09/grad-cam-visualize-class-activation-maps-with-keras-tensorflow-and-deep-learning/
- https://medium.com/analytics-vidhya/visualizing-activation-heatmaps-using-tensorflow-5bdba018f759

Found a related link on TF discussion forum mentioning that this is not possible as of 2021.

- https://discuss.tensorflow.org/t/applying-grad-cam-to-a-model-that-was-converted-to-tensorflow-lite/3810/7

Can you please provide a sample code to execute this
OR guide me with a suitable documentation.
- To model.summary() equivalent for tflite
- Code to visualize last CNN feature map

**Comment by bhack** [comment]
I think that you could try to post directly in that forum thread.

**Comment by mohantym** [comment]
Hi @ra9hur !
You can see the model details of lite model using [netron web app](https://netron.app/). To print all tensor details, you need to invoke the interpreter with respective TFlite model and get the [tensor details](https://www.tensorflow.org/api_docs/python/tf/lite/Interpreter#get_tensor_details) by index of respective tensor.

On GradCam Visualization perspective, You can use @tf.function and tape gradient according to  [On device training documentation](https://www.tensorflow.org/lite/examples/on_device_training/overview) in this [thread](https://discuss.tensorflow.org/t/applying-grad-cam-to-a-model-that-was-converted-to-tensorflow-lite/3810/16?u=manas_mohanty). 

Please post on the same TF Forum  thread if you are looking for more insights on the implementation.

Thank you!

**Comment by ra9hur** [comment]
@mohantym 
Thanks for the suggestion. Haven't got a chance to try it out, will revert for any help.
Thanks again !!

**Comment by ra9hur** [comment]
Sure, thanks again !!

**Comment by mohantym** [comment]
@ra9hur !
Moving this to resolved status then.
Thank you!
